pkg/store/bucket.go

```diff
@@ -1284,6 +1302,15 @@ func newBucketBlock(
 		enablePostingsCompression: enablePostingsCompression,
 	}
 
+	// Translate the block's labels and inject the block ID as a label
+	// to allow to match blocks also by ID.
+	b.matcherLabels = labels.FromMap(meta.Thanos.Labels)
+	b.matcherLabels = append(b.matcherLabels, labels.Label{
```

Comment:
Feels a bit weird to me, what about:
```suggestion
	b.matcherLabels = append(labels.FromMap(meta.Thanos.Labels), labels.Label{
```